src/Controller/Component/AuthorizationComponent.php

```diff
@@ -59,6 +59,30 @@ class AuthorizationComponent extends Component
      * @throws \Authorization\Exception\ForbiddenException when policy check fails.
      */
     public function authorize($resource, $action = null)
+    {
+        if ($action === null) {
+            $request = $this->getController()->request;
```

Comment:
Should use ->getRequest(), ->request is deprecated.

Reply:
This plugin is still using 3.5?

Reply:
IMO those willing to put in the effort to upgrade there existing projects to using the plugin would also be upgrading to latest cake version too.

Reply:
I would like to go to 3.6 soon but we have a ton of plugins that need to upgrade as well. :(

Reply:
I think we should stay 3.5+ for a while longer.